### PR TITLE
Updated spine runtimes, audio fixes

### PR DESCRIPTION
**Pull request by piotrros**
updated spine runtimes
a "fix" for issue #18342 (512k limit should be fine for now)
fixed audio bug from issue #17387, added a test case as well

### CODE REVIEW

**Comment by piotrros** [comment]
Dunno why these automatic builds failed, please help @minggo :)

File: build/cocos2d_tests.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -8372,7 +8375,7 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				CODE_SIGN_IDENTITY = "iPhone Developer";
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = H2S6TV5WJW;
```
Comment (mars3142) [inline build/cocos2d_tests.xcodeproj/project.pbxproj]: I think, you should remove your development from the project.

Comment (piotrros) [reply build/cocos2d_tests.xcodeproj/project.pbxproj]: Fixed, thanks :)

**Comment by minggo** [comment]
Please don't change source file mode to `100755`, and please don't mix  two tasks into one PR.

**Comment by piotrros** [comment]
Sorry, I didn't change file modes intentionally :D Can we pass it this time :)

**Comment by minggo** [comment]
Could you please break this PR into two:

* update Spine runtime
* fix audio issue

**Comment by piotrros** [comment]
Ok:

https://github.com/cocos2d/cocos2d-x/pull/18382

https://github.com/cocos2d/cocos2d-x/pull/18383

**Pull request closed without merging**